poetry/core/masonry/builders/builder.py

```diff
@@ -127,6 +141,15 @@ def find_files_to_add(self, exclude_build=True):  # type: (bool) -> list
                     continue
 
                 if file.is_dir():
+                    if self.format == "sdist" or self.format in include.formats:
```

Comment:
How does this impact cases where include is specified only for wheel? We might need to add more test coverage for edge cases here.

Reply:
The `self.format == "sdist"` check exists because of the cases when inline tables are not used. For example:

```toml
include = ["some.txt", "foo", "bar"]
```

Because they are not specified with a format, it is assumed that they are to be included for sdists.

Regarding specifying wheels only, you are right. Wheels are not affected by this change since the adding mechanism is elsewhere (in `Wheel._copy_module`) which means that if only wheels are specified in `include`, any dir includes are excluded.